great_expectations/datasource/fluent/interfaces.py

```diff
@@ -559,6 +559,14 @@ def _save_context_project_config(self):
             except TypeError as type_err:
                 warnings.warn(str(type_err), GxSerializationWarning)
 
+    def _rebuild_asset_data_connectors(self) -> None:
+        """If Datasource required a data_connector we need to build the data_connector for each asset"""
+        if self.data_connector_type:
+            for data_asset in self.assets:
+                # check if data_connector exist before rebuilding?
+                connect_options = getattr(data_asset, "connect_options", {})
```

Comment:
Any chance we can leverage a type here? Is there a particular subclass of asset that has connect_options? Or do we need to use `getattr` because it doesn't neatly fit into an inheritance hierarchy

Reply:
Yeah, `connect_options` is a convention we've used for our DataAssets that require a `data_connector` but it's not yet a formal part of any interface, so I don't want this to break if a new `DataAsset` is defined that doesn't have a `connect_options` attribute.